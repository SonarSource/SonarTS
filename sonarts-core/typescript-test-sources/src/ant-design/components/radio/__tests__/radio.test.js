import React from 'react';
import { shallow, render } from 'enzyme';
import Radio from '../radio';

describe('Radio', () => {
  it('should render correctly', () => {
    const wrapper = render(<Radio className="customized">Test</Radio>);
    expect(wrapper).toMatchSnapshot();
  });

  it('responses hover events', () => {
    const onMouseEnter = jest.fn();
    const onMouseLeave = jest.fn();

    const wrapper = shallow(
      <Radio
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      />
    );

    wrapper.simulate('mouseenter');
    expect(onMouseEnter).toHaveBeenCalled();

    wrapper.simulate('mouseleave');
    expect(onMouseLeave).toHaveBeenCalled();
  });
});
