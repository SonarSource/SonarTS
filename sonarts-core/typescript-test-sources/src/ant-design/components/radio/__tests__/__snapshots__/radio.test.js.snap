// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `
<label
  class="customized ant-radio-wrapper"
>
  <span
    class="ant-radio"
  >
    <input
      class="ant-radio-input"
      type="radio"
    />
    <span
      class="ant-radio-inner"
    />
  </span>
  <span>
    Test
  </span>
</label>
`;
